tests/integration_tests/test_model_save_and_load.py

```diff
@@ -249,3 +252,212 @@ def test_model_weights_match_training(tmpdir, csv_filename):
+    results_dir = tmp_path / "results"
+    results_dir.mkdir()
+
+    # perform initial model training
+    backend = LocalTestBackend()
+    ludwig_model1 = LudwigModel(config, backend=backend)
+    _, _, output_dir = ludwig_model1.train(
+        training_set=training_set,
+        validation_set=validation_set,
+        test_set=test_set,
+        output_directory="results",  # results_dir
+    )
+
+    # confirm that pretrained model weight was downloaded
+    assert len([f for f in os.listdir(os.path.join(tv_cache_dir, "hub", "checkpoints"))]) == 1
+
+    preds_1, _ = ludwig_model1.predict(dataset=validation_set)
+
+    def check_model_equal(ludwig_model2):
+        # Compare model predictions
+        preds_2, _ = ludwig_model2.predict(dataset=validation_set)
+        assert set(preds_1.keys()) == set(preds_2.keys())
+        for key in preds_1:
+            assert preds_1[key].dtype == preds_2[key].dtype, key
+            assert np.all(a == b for a, b in zip(preds_1[key], preds_2[key])), key
+            # assert preds_2[key].dtype == preds_3[key].dtype, key
+            # assert list(preds_2[key]) == list(preds_3[key]), key
+
+        # Compare model weights
+        for if_name in ludwig_model1.model.input_features:
+            if1 = ludwig_model1.model.input_features[if_name]
+            if2 = ludwig_model2.model.input_features[if_name]
+            for if1_w, if2_w in zip(if1.encoder_obj.parameters(), if2.encoder_obj.parameters()):
+                assert torch.allclose(if1_w, if2_w)
+
+        c1 = ludwig_model1.model.combiner
+        c2 = ludwig_model2.model.combiner
+        for c1_w, c2_w in zip(c1.parameters(), c2.parameters()):
+            assert torch.allclose(c1_w, c2_w)
+
+        for of_name in ludwig_model1.model.output_features:
+            of1 = ludwig_model1.model.output_features[of_name]
+            of2 = ludwig_model2.model.output_features[of_name]
+            for of1_w, of2_w in zip(of1.decoder_obj.parameters(), of2.decoder_obj.parameters()):
+                assert torch.allclose(of1_w, of2_w)
+
+    # remove tv model cache for reloading of Ludwig model
+    shutil.rmtree(tv_cache_dir, ignore_errors=True)
+
+    ludwig_model1.save(tmpdir)
+    ludwig_model_loaded = LudwigModel.load(tmpdir, backend=backend)
+
+    # confirm that torchvision model was not downloaded again
+    assert not os.path.isdir(tv_cache_dir)
+
+    # confirm model structure and weights are the same
+    check_model_equal(ludwig_model_loaded)
+
+    # Test loading the model from the experiment directory
+    ludwig_model_exp = LudwigModel.load(os.path.join(output_dir, "model"), backend=backend)
+
+    # confirm that torchvision model was not downloaded again
+    assert not os.path.isdir(tv_cache_dir)
+
+    # confirm model structure and weights are the same
+    check_model_equal(ludwig_model_exp)
+
+
+@pytest.mark.skip(
+    reason="(jimthompson5802)Pending decision on need for this test and how to test for downloading of weights"
```

Comment:
How big is the weights file? Does this test run slowly?

We have unit tests that download weights from huggingface for text encoders that rarely seem to run into issues. However, if the test runs slowly, I would lean towards marking this test as `@slow` and skipping it by default.

Reply:
Locally on my MBP, the test downloads about 440 MB weights and runs about 100 seconds.  I'll reenable it and if needed mark it `@slow` later.

Reply:
@justinxzhao  One complication with this.  The test works without issue when I run it locally on my development environment.  However, it fails in the GH Actions CI Test run.  For the moment, I've re-marked this to be skipped.  There appears to be some kind of environment setup difference from my local environment and the setup for the GH Actions. 

The error occurs in how the `assert` test confrims correct operation and not with the core Ludwig or Hugging Face component.  From this perspective, this **does not** affect Ludwig functionality.  Given  the test works locally,  the issue appears to be more around how the test was designed to operate. I have to dig into this.

Reply:
Simplified the test.